src/Totem/Services/TesterService.cs

```diff
@@ -194,6 +199,32 @@ public TestMessageResult DoAllMessageValuesMatchDataTypes(CaseInsensitiveDiction
             }
         }
 
+        private static void CheckNumberType(SchemaObject propertySchemaObject, KeyValuePair<string, object> kv,
+            TestMessageResult testMessageResult)
+        {
+            var isDouble = double.TryParse(kv.Value.ToString(), out _);
+            var isFloat = float.TryParse(kv.Value.ToString(), out _);
+            
+            // Validate number data type
+            if (!isDouble && !isFloat)
+            {
+                AddTypeError(testMessageResult, kv.Value.ToString(), kv.Key, propertySchemaObject.Type);
+            }
+
+            if (propertySchemaObject.Format != null)
+            {
+                // Validate specific number formats
+                if (propertySchemaObject.Format.EqualsCaseInsensitive("float") && !isFloat)
```

Comment:
**SHOULD:** Also here

Reply:
Resolved in 5ad523f0ac21a9827e4ffdaac83a3a41ec6c9ff6